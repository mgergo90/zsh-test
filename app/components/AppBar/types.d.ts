import { ChangeEvent } from 'react';

export interface AppBarProps {
  onChange: (event: ChangeEvent<HTMLInputElement>) => void;
}
