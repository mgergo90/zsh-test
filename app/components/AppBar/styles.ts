import {
  fade,
  makeStyles,
  Theme,
  createStyles,
} from '@material-ui/core/styles';

export const useAppBarStyle = makeStyles((theme: Theme) =>
  createStyles({
    grow: {
      flexGrow: 1,
    },
    search: {
      position: 'relative',
      borderRadius: theme.shape.borderRadius,
      backgroundColor: fade(theme.palette.common.white, 0.15),
      '&:hover': {
        backgroundColor: fade(theme.palette.common.white, 0.25),
      },
      marginRight: theme.spacing(2),
      marginLeft: 0,
      width: '100%',
      [theme.breakpoints.up('sm')]: {
        width: 'auto',
      },
    },
    searchIcon: {
      padding: theme.spacing(0, 2),
      height: '100%',
      position: 'absolute',
      pointerEvents: 'none',
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'center',
    },
    inputRoot: {
      color: 'inherit',
    },
    toolbar: {
      paddingLeft: 0,
    },
    inputInput: {
      padding: theme.spacing(1, 1, 1, 0),
      paddingLeft: `calc(1em + ${theme.spacing(4)}px)`,
      transition: theme.transitions.create('width'),
      width: '100%',
      [theme.breakpoints.up('md')]: {
        width: '20ch',
      },
    },
  }),
);
