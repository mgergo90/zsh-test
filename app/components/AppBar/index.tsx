import React from 'react';
import SearchIcon from '@material-ui/icons/Search';
import { Container, Toolbar, InputBase, AppBar } from '@material-ui/core';

import { useAppBarStyle } from './styles';
import { AppBarProps } from './types';

const ZshAppBar = ({ onChange }: AppBarProps) => {
  const classes = useAppBarStyle();

  return (
    <div className={classes.grow}>
      <AppBar position="static">
        <Container maxWidth="lg">
          <Toolbar className={classes.toolbar}>
            <div className={classes.search}>
              <div className={classes.searchIcon}>
                <SearchIcon />
              </div>
              <InputBase
                placeholder="Searchâ€¦"
                onChange={onChange}
                classes={{
                  root: classes.inputRoot,
                  input: classes.inputInput,
                }}
                inputProps={{ 'aria-label': 'search' }}
              />
            </div>
            <div className={classes.grow} />
          </Toolbar>
        </Container>
      </AppBar>
    </div>
  );
};

export default ZshAppBar;
