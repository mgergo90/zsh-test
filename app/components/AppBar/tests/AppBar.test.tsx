import React from 'react';
import { createRenderer } from 'react-test-renderer/shallow';
import { render, fireEvent, screen } from '@testing-library/react';

import AppBar from '../';

const renderer = createRenderer();

describe('<AppBar />', () => {
  it('should render and match the snapshot', () => {
    renderer.render(<AppBar onChange={jest.fn()} />);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
  it('should call onchange', () => {
    const onChange = jest.fn();
    const event = {
      target: { value: 'find movie' },
    };
    render(<AppBar onChange={onChange} />);
    fireEvent.change(screen.getByPlaceholderText('Searchâ€¦'), event);
    expect(onChange).toHaveBeenCalledTimes(1);
  });
});
