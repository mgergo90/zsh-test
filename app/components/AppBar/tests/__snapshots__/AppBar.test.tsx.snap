// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppBar /> should render and match the snapshot 1`] = `
<div
  className="makeStyles-grow-1"
>
  <WithStyles(ForwardRef(AppBar))
    position="static"
  >
    <WithStyles(ForwardRef(Container))
      maxWidth="lg"
    >
      <WithStyles(ForwardRef(Toolbar))
        className="makeStyles-toolbar-5"
      >
        <div
          className="makeStyles-search-2"
        >
          <div
            className="makeStyles-searchIcon-3"
          >
            <Memo(SearchIcon) />
          </div>
          <WithStyles(ForwardRef(InputBase))
            classes={
              Object {
                "input": "makeStyles-inputInput-6",
                "root": "makeStyles-inputRoot-4",
              }
            }
            inputProps={
              Object {
                "aria-label": "search",
              }
            }
            onChange={[MockFunction]}
            placeholder="Searchâ€¦"
          />
        </div>
        <div
          className="makeStyles-grow-1"
        />
      </WithStyles(ForwardRef(Toolbar))>
    </WithStyles(ForwardRef(Container))>
  </WithStyles(ForwardRef(AppBar))>
</div>
`;
